Task completed: GP-127. Controls: Keyboard

frnzee opened the pull request:
- added "new" Unity InputSystem package
- built input actions map
- added controls manager and debug.log entities for clarity it's working

Jira task: https://lighteatersstudio.atlassian.net/browse/GP-127?atlOrigin=eyJpIjoiZjI3N2M2ZTc4ZDk0NDkzNWJiN2UzMjk5YmU4ZWVmZGUiLCJwIjoiaiJ9

frnzee committed:
Added controls handler

- added "new" Unity InputSystem package
- built input actions map
- added controls manager and debug.log entities for clarity it's working

frnzee committed:
Added namespace

- added namespace and using this namespace in game installer

ega-ega commented on Assets/Prefabs/Gameplay/PlayerInputActionsMap.inputactions:
Move this file to folder: Gameplay/Input

frnzee replied:
Done

frnzee committed:
Input controller rework

- added interface IInputController
- fixed naming
- removed debug entities
- removed MonoBehaviour from InputController

ega-ega commented on Assets/Scripts/Services/Input/InputController.cs:
```diff
@@ -0,0 +1,66 @@
+
+        private void Move(InputAction.CallbackContext ctx)
+        {
+            MoveChanged?.Invoke(Vector2.zero);
```
Pls change zero vector to specific value from ctx parametr

frnzee replied:
Done

ega-ega commented on Assets/Scripts/Services/Input/InputController.cs:
```diff
@@ -0,0 +1,66 @@
+
+        private void Fire(InputAction.CallbackContext ctx)
+        {
+            FireChanged?.Invoke(Vector2.zero);
```
Pls change zero vector to specific value from ctx parametr

frnzee replied:
Done

ega-ega commented on Assets/Scripts/Services/Input/InputController.cs:
```diff
@@ -0,0 +1,66 @@
+
+        private void Special(InputAction.CallbackContext ctx)
+        {
+            SpecialChanged?.Invoke(Vector2.zero);
```
Pls change zero vector to specific value from ctx parametr

frnzee replied:
Done

ega-ega commented on Assets/Scripts/Services/Input/InputController.cs:
```diff
@@ -0,0 +1,66 @@
+            OnEnable();
+        }
+
+        private void Move(InputAction.CallbackContext ctx)
```
Pls rename parametr to context. By our agreement, the use of abbreviations or acronyms must not be allowed. as an exception to this rule, we can take the name "UI" or "FX", which is currently recognized as a concept and not an acronym by the vast majority of people / programmers

frnzee replied:
Done

ega-ega commented on Assets/Scripts/Services/Input/InputController.cs:
```diff
@@ -0,0 +1,66 @@
+            MoveChanged?.Invoke(Vector2.zero);
+        }
+
+        private void Fire(InputAction.CallbackContext ctx)
```
Pls rename parametr to context. By our agreement, the use of abbreviations or acronyms must not be allowed. as an exception to this rule, we can take the name "UI" or "FX", which is currently recognized as a concept and not an acronym by the vast majority of people / programmers

frnzee replied:
Done

ega-ega commented on Assets/Scripts/Services/Input/InputController.cs:
```diff
@@ -0,0 +1,66 @@
+            FireChanged?.Invoke(Vector2.zero);
+        }
+
+        private void Special(InputAction.CallbackContext ctx)
```
Pls rename parametr to context. By our agreement, the use of abbreviations or acronyms must not be allowed. as an exception to this rule, we can take the name "UI" or "FX", which is currently recognized as a concept and not an acronym by the vast majority of people / programmers

frnzee replied:
Done

ega-ega commented on Assets/Scripts/Services/Input/InputController.cs:
```diff
@@ -0,0 +1,66 @@
+            SpecialChanged?.Invoke(Vector2.zero);
+        }
+
+        private void Melee(InputAction.CallbackContext ctx)
```
Pls rename parametr to context. By our agreement, the use of abbreviations or acronyms must not be allowed. as an exception to this rule, we can take the name "UI" or "FX", which is currently recognized as a concept and not an acronym by the vast majority of people / programmers

frnzee replied:
Done

ega-ega commented on Assets/Scripts/Services/Input/InputController.cs:
```diff
@@ -0,0 +1,66 @@
+            MeleeChanged?.Invoke();
+        }
+
+        private void Use(InputAction.CallbackContext ctx)
```
Pls rename parametr to context. By our agreement, the use of abbreviations or acronyms must not be allowed. as an exception to this rule, we can take the name "UI" or "FX", which is currently recognized as a concept and not an acronym by the vast majority of people / programmers

frnzee replied:
Done

ega-ega commented on Assets/Scripts/Services/Input/InputController.cs:
```diff
@@ -0,0 +1,66 @@
+            UseChanged?.Invoke();
+        }
+
+        private void Reload(InputAction.CallbackContext ctx)
```
Pls rename parametr to context. By our agreement, the use of abbreviations or acronyms must not be allowed. as an exception to this rule, we can take the name "UI" or "FX", which is currently recognized as a concept and not an acronym by the vast majority of people / programmers

frnzee replied:
Done

ega-ega commented on Assets/Scripts/Services/Input/InputController.cs:
```diff
@@ -0,0 +1,66 @@
+
+        private void OnEnable()
+        {
+            _inputActionAsset.FindAction("Move").performed += Move;
```
Pls extract string value to const.By our convention, your implementation falls under the magic number/value rule.
Also select the rest of the strings as constants

frnzee replied:
Done

ega-ega commented on Assets/Scripts/Services/Input/InputController.cs:
```diff
@@ -0,0 +1,66 @@
+            OnEnable();
+        }
+
+        private void Move(InputAction.CallbackContext ctx)
```
Rename Method to OnMove. Also pls rename all another handlers in this class

frnzee replied:
Done

frnzee committed:
Refactoring

- fixed naming
- strings moved to consts
- vector2.zero parameters changed to vector2 values from context

ega-ega commented on Assets/Scripts/Installer/GameInstaller.cs:
```diff
@@ -24,8 +26,11 @@ public class GameInstaller : MonoInstaller
+
+        [Header("Input")]
+        [SerializeField] private InputActionAsset _playerInputActionsMap;
+        private InputController _playerInputController;
```
Pls remove unused field

frnzee committed:
Refactoring

- removed unused field

frnzee committed:
Refactoring

- resolved problems with develop branch

frnzee committed:
Merge branch 'develop' into feature/GP-127_Controls
